prog/kube-utils/main.go

```diff
@@ -183,6 +190,42 @@ func reclaimPeer(weave weaveClient, cml *configMapAnnotations, peerName string,
 	return true, err
 }
 
+// regiesters with Kubernetes API server for node delete events. Node delete event handler
+// invokes reclaimRemovedPeers to remove it from IPAM so that IP space is reclaimed
+func registerForNodeUpdates(client *kubernetes.Clientset, stopCh <-chan struct{}, nodeName, peerName string) {
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+	common.Log.Debugln("registering for updates for node delete events")
+	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: func(obj interface{}) {
+			node := obj.(*v1core.Node)
```

Comment:
I think this can be a `cache.DeletedFinalStateUnknown`, which is possible after this process has reconnected to the api-server.

Reply:
Added to logic to handle deleted objects.